packages/lib/src/vis/shared/ViewportCenterer.tsx

```diff
@@ -0,0 +1,29 @@
+import { useFrame, useThree } from '@react-three/fiber';
+import { useEffect, useRef } from 'react';
+import type { Vector2 } from 'three';
+
+import { useAxisSystemContext } from './AxisSystemContext';
+import { useMoveCameraTo } from './hooks';
+
+function ViewportCenterer() {
+  const { dataToWorld, worldToData } = useAxisSystemContext();
+  const viewportCenter = useRef<Vector2>();
+  const { position } = useThree((state) => state.camera);
+
+  const moveCameraTo = useMoveCameraTo();
+
+  useFrame(() => {
+    viewportCenter.current = worldToData(position);
```

Comment:
Rather than computing the new camera position in interaction meshes and store the computed position in the callback, the camera position (in data coordinates) is now saved at each frame.